manifests/agent.pp

```diff
@@ -220,6 +223,8 @@
   $zabbix_version                                 = $zabbix::params::zabbix_version,
   $zabbix_package_state                           = $zabbix::params::zabbix_package_state,
   $zabbix_package_agent                           = $zabbix::params::zabbix_package_agent,
+  $zabbix_package_provider                        = $zabbix::params::zabbix_package_provider,
```

Comment:
Do we need to set ther provider explicity? Doesn't puppet pick the correct one on Windows? If it's required, please use a proper datatype like:
```suggestion
  Optional[String[1]] $zabbix_package_provider    = $zabbix::params::zabbix_package_provider,
```

Reply:
hello,
It seems the default value for windows is "windows", MSI or Exe files. Source => https://puppet.com/docs/puppet/5.5/types/package.html#package-provider-windows.

I can add a condition like below ?
```
if $::kernel == 'windows' {
  package { $zabbix_package_agent:
    ensure   => $zabbix_package_state,
    require  => Class['zabbix::repo'],
    provider => $zabbix_package_provider,
    tag      => 'zabbix',
  }
 } else {
  # Installing the package
  package { $zabbix_package_agent:
    ensure   => $zabbix_package_state,
    require  => Class['zabbix::repo'],
    tag      => 'zabbix',
  }
```

Reply:
I think the parameter is the right way to go. I was just hoping that puppet uses choco by default.